### PR TITLE
Add Icon to gutter for wizard and unbreak wizard

### PR DESCRIPTION
**Pull request by paulvanbrenk**
Adding the icon back to the wizard, and fix the broken wizard

### CODE REVIEW

**Comment by paulvanbrenk** [comment]
image

File: Nodejs/Product/ProjectWizard/ImportWizard.cs [hunk 1]
```diff
@@ -15,74 +14,75 @@ namespace Microsoft.NodejsTools.ProjectWizard
...
-            if (dte == null)
+            var provider = automationObject as Microsoft.VisualStudio.OLE.Interop.IServiceProvider;
+            if (provider == null)
             {
-                var provider = automationObject as Microsoft.VisualStudio.OLE.Interop.IServiceProvider;
-                if (provider != null)
-                {
-                    dte = new ServiceProvider(provider).GetService(typeof(DTE)) as DTE;
-                }
+                MessageBox.Show(ProjectWizardResources.ErrorNoDte, SR.ProductName);
```
Comment (amcasey) [inline Nodejs/Product/ProjectWizard/ImportWizard.cs]: Is this the appropriate message?

Comment (amcasey) [reply Nodejs/Product/ProjectWizard/ImportWizard.cs]: Oh, the resource just has a confusing name.

File: Nodejs/Product/ProjectWizard/ImportWizard.cs [hunk 2]
```diff
@@ -15,74 +14,75 @@ namespace Microsoft.NodejsTools.ProjectWizard
...
-                Guid packageGuid = new Guid(Guids.NodejsPackageString);
-                IVsPackage package;
-                ((IVsShell)Package.GetGlobalService(typeof(SVsShell))).LoadPackage(
-                    ref packageGuid,
-                    out package
-                );
+                var hr = EnsurePackageLoaded(serviceProvider);
+                if (ErrorHandler.Failed(hr))
+                {
+                    MessageBox.Show(string.Format(ProjectWizardResources.ImportWizardCouldNotStartFailedToLoadPackage, hr), "Visual Studio");
```
Comment (amcasey) [inline Nodejs/Product/ProjectWizard/ImportWizard.cs]: Should `"Visual Studio"` come from a resource (or at least a constant) so we don't have to keep them all in sync?

File: Nodejs/Product/ProjectWizard/ImportWizard.cs [hunk 3]
```diff
@@ -15,74 +14,75 @@ namespace Microsoft.NodejsTools.ProjectWizard
...
+                    throw new WizardBackoutException();
+                }
+                var uiShell = (IVsUIShell)serviceProvider.GetService(typeof(SVsUIShell));
 
-                System.Threading.Tasks.Task.Factory.StartNew(() =>
+                var projName = replacementsDictionary["$projectname$"];
+                replacementsDictionary.TryGetValue("$specifiedsolutionname$", out var solnName);
```
Comment (amcasey) [inline Nodejs/Product/ProjectWizard/ImportWizard.cs]: I don't think you can assume that this sets null on failure.  I'd check the return value in the `if` below.

File: Nodejs/Product/ProjectWizard/ImportWizard.cs [hunk 4]
```diff
@@ -15,74 +14,75 @@ namespace Microsoft.NodejsTools.ProjectWizard
...
+                string directory;
+                if (string.IsNullOrWhiteSpace(solnName))
+                {
+                    // Create directory is unchecked, destinationdirectory is the
```
Comment (amcasey) [inline Nodejs/Product/ProjectWizard/ImportWizard.cs]: I'm not sure what this comment means.

File: Nodejs/Product/ProjectWizard/ImportWizard.cs [hunk 5]
```diff
@@ -15,74 +14,75 @@ namespace Microsoft.NodejsTools.ProjectWizard
...
-                    {
-                        // Create directory is checked, the destinationdirectory is the
-                        // directory the user entered plus the project name plus the
-                        // solution name.
-                        directory = Path.GetDirectoryName(replacementsDictionary["$destinationdirectory$"]);
-                    }
+                    // Create directory is checked, the destinationdirectory is the
+                    // directory the user entered plus the project name plus the
+                    // solution name - we want to remove both extra folders
+                    directory = Path.GetDirectoryName(Path.GetDirectoryName(replacementsDictionary["$destinationdirectory$"]));
```
Comment (amcasey) [inline Nodejs/Product/ProjectWizard/ImportWizard.cs]: Presumably, there's some reason to believe `$destinationdirectory$` has a grandparent?

File: Nodejs/Product/Nodejs/Nodejs.csproj [hunk 6]
```diff
@@ -679,6 +679,36 @@
     <ZipProject Include="ProjectTemplates\NodejsWebApp\Preview.png" />
     <ZipProject Include="ProjectTemplates\AzureNodejsApp\Preview.png" />
     <ZipProject Include="ProjectTemplates\NodejsConsoleApp\Preview.png" />
+    <Resource Include="..\Icons\NewProjectDialogIconsICO\JSAzureApplication.ico">
```
Comment (amcasey) [inline Nodejs/Product/Nodejs/Nodejs.csproj]: Where are these consumed?

**Comment by amcasey** [comment]
Context for other reviewers: this is a port of a fix from PTVS.

**Pull request merged**